Fw/Types/StringUtils.cpp

```diff
@@ -1,7 +1,16 @@
 #include "StringUtils.hpp"
+#include <Fw/Types/Assert.hpp>
 #include "string.h"
 
 char* Fw::StringUtils::string_copy(char* destination, const char* source, U32 num) {
+    // Handle self-copy
+    if(destination == source) {
+        return destination;
+    }
+
+    // Copying an overlapping range is undefined
+    FW_ASSERT(source < destination || destination + num - 1 < source);
```

Comment:
I am not sure this is a complete check.  Consider source = destination -1 and num = 16, it will quickly violate the overlapping rule.

Perhaps:
```
FW_ASSERT(source + num <= destination || distination + num <= source);
```